File: COntest]/Heap or not/Problem.cpp
string isMinHeap(int A[], int N) {
    for (int i = 0; i <= (N - 2) / 2; ++i) {
        int left = 2 * i + 1;
        int right = 2 * i + 2;

        if (left < N && A[i] > A[left]) {
            return "No";
        }
        if (right < N && A[i] > A[right]) {
            return "No";
        }
    }
    return "Yes";
}

File: Codechef/Contest/Subarray with k distinct number/Problem.py
class Solution:
    def subarraysWithKDistinct(self, nums, k):
        def atMostK(arr, k):
            count = {}
            res = left = 0
            for right in range(len(arr)):
                count[arr[right]] = count.get(arr[right], 0) + 1
                while len(count) > k:
                    count[arr[left]] -= 1
                    if count[arr[left]] == 0:
                        del count[arr[left]]
                    left += 1
                res += right - left + 1
            return res

        return atMostK(nums, k) - atMostK(nums, k - 1)